ssh-krb-node-executor-plugin/contents/scp-krb-executor.sh

```diff
@@ -13,36 +13,41 @@ shift
 SRC_FILE=$1
 shift
 DEST_FILE=$1
-KERB_KEYTAB=$RD_CONFIG_KERBEROS_KEYTAB
 
-if [ ! -f "$KERB_KEYTAB" ]; then
-  echo "Keytab $KERB_KEYTAB not found" >&2
+
+if [ ! -f "$RD_CONFIG_KERBEROS_KEYTAB" ]; then
+  echo "Keytab $RD_CONFIG_KERBEROS_KEYTAB not found" >&2
   exit 2
 fi
 
 # random delay (0..0.5s) to make it work with parallel exec
 sleep "$(bc <<< "scale=2; $(printf '0.%02d' $(( RANDOM % 100))) / 2")"
 
-# status 1 if the credentials cache cannot be read or is expired, and with status 0 otherwise
-klist -s
-if [ $? -ne 0 ]; then
-  kinit -kt "$KERB_KEYTAB" "$RD_CONFIG_KERBEROS_USER"
 
-  # verify ticket has been successfuly created
-  klist -s
-  if [ $? -ne 0 ]; then
-    echo "Kinit failure when calling kinit -kt $KERB_KEYTAB $RD_CONFIG_KERBEROS_USER" >&2
+# override the default kerberos cache file if required
+if [[ "$RD_CONFIG_USE_KERBEROS_CUSTOM_CACHE_FILE" == 'true' ]]; then
+  if [ -n "$RD_CONFIG_KERBEROS_CUSTOM_CACHE_FILENAME" ]; then
+    export KRB5CCNAME="$RD_CONFIG_KERBEROS_CUSTOM_CACHE_FILENAME"
+  else
+    export KRB5CCNAME="/tmp/krb5cc_$(id -ru)_rundeck_$RD_CONFIG_KERBEROS_USER"
+  fi
+fi
+
+
+# recreate the cache if it's expired or not present
+if ! klist -s; then
+  kinit -kt "$RD_CONFIG_KERBEROS_KEYTAB" "$RD_CONFIG_KERBEROS_USER"
+  if ! klist -s; then
+    echo "Kinit failure when calling kinit -kt $RD_CONFIG_KERBEROS_KEYTAB $RD_CONFIG_KERBEROS_USER" >&2
     exit 2
   fi
 fi
 
 SSHOPTS=" -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o LogLevel=quiet -o GSSAPIDelegateCredentials=yes"
-RUNSSH="scp $SSHOPTS $SRC_FILE $USER@$HOST:$DEST_FILE"
+RUNSSH='scp $SSHOPTS "$SRC_FILE" $USER@$HOST:"${DEST_FILE// /\\ }"'
 
 #finally, execute scp but don't print to STDOUT
-$RUNSSH 1>&2 || exit $? # exit if not successful
+eval $RUNSSH 1>&2 || exit $? # exit if not successful
 echo "$DEST_FILE" # echo remote filepath
 
-if [[ ${RD_CONFIG_DO_KDESTROY} == 'true' ]]; then
-  kdestroy
-fi
+[[ "$RD_CONFIG_DO_KDESTROY" == 'true' ]] && kdestroy
```

Comment:
Whereas I like this change, it's unrelated. Same background as https://github.com/cernops/rundeck-exec-kerberos/pull/7#discussion_r970721893. I'm happy to take separate change requests if you're still motivated to make the code look nicer after all this nitpicking :)